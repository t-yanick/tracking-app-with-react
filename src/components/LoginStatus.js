import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const LoginStatus = ({ loginStats, user }) => (
  <div className="login-status">
    {loginStats ? `Welcome, ${user.user.username}` : 'Please log in.'}
  </div>
);

const mapStateToProps = (state) => ({
  loginStats: state.user.logIn,
  user: state.user,
});

LoginStatus.propTypes = {
  user: PropTypes.instanceOf(Object),
  loginStats: PropTypes.bool,
};

LoginStatus.defaultProps = {
  user: {},
  loginStats: false,
};

export default connect(mapStateToProps)(LoginStatus);
