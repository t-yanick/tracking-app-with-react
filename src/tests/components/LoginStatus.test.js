import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import LoginStatus from '../../components/LoginStatus';
import createTestStore from '../fixtures/createTestStore';

describe('LoginStatus', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <LoginStatus />
          </MemoryRouter>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
