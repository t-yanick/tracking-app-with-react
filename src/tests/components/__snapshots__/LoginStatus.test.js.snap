// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginStatus should match with snapshot 1`] = `
<div
  className="login-status"
>
  Please log in
</div>
`;
